sol/Usdx.sol

```diff
@@ -37,12 +38,19 @@ contract USDX is ERC20, Ownable {
 	function setFeed(address _newFeed) public onlyOwner {
 		AggregatorV3Interface newFeed = AggregatorV3Interface(_newFeed);
 		require(newFeed.decimals() == FEED_DECS);
+
+		(,,,,uint80 answeredInRound) = newFeed.latestRoundData();
+		(,,,uint256 updatedAt,) = newFeed.getRoundData(answeredInRound);
+		uint256 difference = uint256(block.timestamp).sub(updatedAt);
+		require(difference < FEED_EXPIRE_TIME, "Price has been updated in last 2 days, unable to set feed.");
+
 		ethUsdPriceFeed = newFeed;
 	}
 
 	receive() external payable {
-		(,int256 rate,,,) = ethUsdPriceFeed.latestRoundData();
+		(uint80 roundId,int256 rate,,,uint80 answeredInRound) = ethUsdPriceFeed.latestRoundData();
 		uint256 usd = msg.value.mul(rate.toUint256()).div(10**FEED_DECS); //latestRoundData uses 8 decimals
+		require(roundId == answeredInRound, "Current price is not from the latest round.");
```

Comment:
From https://docs.chain.link/docs/faq#how-can-i-check-if-the-answer-to-a-round-is-being-carried-over-from-a-previous-round:
> How can I check if the answer to a round is being carried over from a previous round?
> 
> You can check answeredInRound against the current roundId. If answeredInRound is less than roundId, the answer is being carried over. If answeredInRound is equal to roundId, then the answer is fresh.

I don't fully understand the distinction between roundId and answeredInRound, but the documentation above doesn't make it seem like this is necessarily an error condition.

From the release notes https://docs.chain.link/docs/developer-communications:
> TA new function latestRoundData has been added. It returns roundId, answer, startedAt, updatedAt, and answeredInRound. This allows consumers of reference data to make decisions around if the data is acceptable in a single function call (ex: if answeredInRound < roundId could indicate stale data). 

So, this definitely sounds like there is potential for stale data, but the "could" implies that this might not always be the case.

I propose the following.  We check that `require(roundId - answeredInRound <= stalenessThreshold)`, and then set `stalenessThreshold == 0`, but allow it to be modifiable by owner.  This way, if `roundId != answeredInRound` is ever appropriate and/or common, the owner of the contract can change the value to something sensible.